controllers/snapshot/snapshot_suite_test.go

```diff
@@ -99,6 +100,8 @@ var _ = BeforeSuite(func() {
 	k8sClient = k8sManager.GetClient()
 	go func() {
 		defer GinkgoRecover()
+		Expect(cache.SetupBindingApplicationCache(k8sManager)).To(Succeed())
+		Expect(cache.SetupBindingEnvironmentCache(k8sManager)).To(Succeed())
```

Comment:
It should be possible to prevent using cache (which causes issues in low-resource environments) in tests by using mockLoaders for these resources in the adapter test itself, unless the issue is somewhere else?

Reply:
yeah, I found out how to mock it, but I have to rewrite mocker implementation, I need to return nil and it's not compatible

```
	value := ctx.Value(contextKey)
	if value == nil {
		panic("Mocked data not found in the context")
	}
```

Reply:
nevermind, the issue was somewhere else

Reply:
Cool, I can see that it works now, thanks